crates/chia-client/src/request_map.rs

```diff
@@ -0,0 +1,65 @@
+use std::{collections::HashMap, sync::Arc};
+
+use chia_protocol::Message;
+use tokio::sync::{oneshot, Mutex, OwnedSemaphorePermit, Semaphore};
+
+#[derive(Debug)]
+pub struct Request {
+    sender: oneshot::Sender<Message>,
+    _permit: OwnedSemaphorePermit,
+}
+
+impl Request {
+    pub fn send(self, message: Message) {
+        self.sender.send(message).ok();
+    }
+}
+
+#[derive(Debug)]
+pub struct RequestMap {
+    items: Mutex<HashMap<u16, Request>>,
+    semaphore: Arc<Semaphore>,
```

Comment:
`Mutex` is a tokio mutex, to be used with coroutines. `Arc` is the standard library primitive to be used with threads. It seems suspicious to mix these.

Reply:
It would be good to add a comment explaining what the semaphore is used for. It's kind of an odd primitive that was used in the early days, before mutexes and condition variables were invented.

Reply:
why does this need to be held by an `Arc`? The whole `RequestMap` is already held by an `Arc`

Reply:
Because you can't get owned semaphore guards from a non-`Arc` `Semaphore`